src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false) {
+      animal.health -= 50;
+    }
```

Comment:
Please read task one more time. As I remember, you should also check isn't this exemplary Carnivor.

Reply:
hidden exists only in herbivores, I thought I don't have to check type of animal.